src/code/z_player_lib.c

```diff
@@ -710,35 +783,29 @@ void func_8008F470(GlobalContext* globalCtx, void** skeleton, Vec3s* jointTable,
     CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_player_lib.c", 1803);
 }
 
-Vec3f D_8012602C = { 0.0f, 0.0f, 0.0f };
-
-Vec3f D_80126038[] = {
-    { 1304.0f, 0.0f, 0.0f },
-    { 695.0f, 0.0f, 0.0f },
-};
-
-f32 D_80126050[] = { 1265.0f, 826.0f };
-f32 D_80126058[] = { SQ(13.04f), SQ(6.95f) };
-f32 D_80126060[] = { 10.019104f, -19.925102f };
-f32 D_80126068[] = { 5.0f, 3.0f };
+static Vec3f sZeroVec = { 0.0f, 0.0f, 0.0f };
 
-Vec3f D_80126070 = { 0.0f, -300.0f, 0.0f };
-
-void func_8008F87C(GlobalContext* globalCtx, Player* this, SkelAnime* skelAnime, Vec3f* pos, Vec3s* rot,
+void Player_FeetIK(GlobalContext* globalCtx, Player* this, SkelAnime* skelAnime, Vec3f* pos, Vec3s* rot,
                    s32 thighLimbIndex, s32 shinLimbIndex, s32 footLimbIndex) {
```

Comment:
what is IK? I see this term used alot in modding contexts, but since i dont really do modeling stuff I dont know

Reply:
https://en.wikipedia.org/wiki/Inverse_kinematics

ELI5: find rotations that look nice for bones between feet and body given only the position of the feet and the position of the body

Reply:
i feel like theres probably a name we can use that would be easier to understand without needing that context, but i dont have any suggestions for now

Reply:
A brief description about what IK is and what this is doing to achieve that would be an option as well.
I just want to make sure this is actually IK. I cant follow any of that math thats going on, did you verify that what its achieving is accurately described as IK?

Reply:
yeah I called it IK because it handles moving links feet to be in the right spot on stairs etc

Reply:
i guess one thing that could help, would it make more sense to say this updating the feet IK? instead of just *being* IK
the term `FeetIK` on its own doesnt really help tell me what its achieving, even after learning what IK is

Reply:
I've now documented this function, and renamed it to `Player_SnapFootToFloor` to be more general.